tronweb/coffee/graph.coffee

```diff
@@ -1,126 +1,324 @@
+
+        layout = cy.layout(module.GraphUtils.getDefaultLayout())
+        layout.run()
+
+        # Fit all elements (if no elements are specified it fits all, so we pass undefined) with 25px padding
+        cy.fit(undefined, 25)
+        cy.center()
+
+        cy.nodes().forEach (node) ->
+            node.data 'manuallyPositioned', false
+
+    defaultStylesheet: [
+        {
+            selector: 'node',
+            style: {
+                'label': 'data(name)'
+                'text-valign': 'center'
+                'text-halign': 'center'
+                'background-color': '#f8f8f8'
+                'color': '#000'
+                'font-size': '16px'
+                'font-weight': 'bold'
+                'shape': 'roundrectangle'
+                'width': (node) -> module.GraphUtils.getNodeWidth(node)
+                'height': 25
+                'padding-top': '5px'
+                'padding-bottom': '5px'
+                'padding-left': '10px'
+                'padding-right': '10px'
+                'text-wrap': 'none'
+                'border-width': 3
+                'border-color': '#999'
+            }
+        },
+        {
+            selector: 'edge',
+            style: {
+                'curve-style': 'bezier'
+                'target-arrow-shape': 'triangle'
+                'target-arrow-color': '#999'
+                'line-color': '#999'
+                'width': 2
+            }
+        },
+        # This is unfortunately necessary. For consistency, this duplicates some of the colours and styles defined in our LESS.
+        # The LESS classes (.succeeded, .running, etc.) apply to HTML elements, while the styles here apply to the graph's
+        # Cytoscape nodes (SVG canvas elements). Those LESS variables only exist during LESS compilation, so we can't use them here.
+        #
+        # If you change a colour here, you should also change it in the LESS, unless you're trying to be more specific about states
+        # in the graph than the state mixins allow.
+        #
+        # Success states (.success mixin in LESS)
+        {
+            selector: '.succeeded',
+            style: {
+                'border-color': '#218E0B'      # @green from LESS
+                'background-color': '#f0ffe0'  # Light green
+            }
+        },
+        # Info states (.info mixin in LESS)
+        {
+            selector: '.running, .starting',
+            style: {
+                'border-color': '#2F47B8'      # @blue
+                'background-color': '#f0f5ff'  # Light blue
+            }
+        },
+        # Warning states (.warning mixin in LESS)
+        {
+            selector: '.cancelled, .skipped',
+            style: {
+                'border-color': '#A6790D'      # @yellow
+                'background-color': '#fffbf0'  # Light yellow
+            }
+        },
+        # Error states (.error mixin in LESS)
+        {
+            selector: '.failed, .unknown',
```

Comment:
We hightlight "unknown" action states as red now, but our graph is quite dumb so it isn't highlighting the node. I'm a bit worried about doing this in something like an ad pipeline job where crossjob triggers will show as red nodes. Maybe this should be some alternative colour